### PR TITLE
Base32 and CSV added for core:encoding

### PR DESCRIPTION
**Pull request by zhibog**
CSV is pretty much done, unless you would want to add more complexity.

Base32 could probably use some refactoring, as it is based on ref stuff from Go. but it encodes the test vectors correctly.

Also provided the test vectors to the Base64 package for reference.

### CODE REVIEW

File: core/encoding/base64/base64.odin [hunk 1]
```diff
@@ -90,4 +90,39 @@ decode :: proc(data: string, DEC_TBL := DEC_TABLE, allocator := context.allocato
...
+        delete(enc);
+    }
+
+    // Decode test
+    for v in test_vectors {
+        dec := decode(v.encoded);
+        fmt.printf("decode(\"%s\") => \"%s\" \t| want: \"%s\"\n", v.encoded, string(dec), v.plain);
+        delete(dec);
+    }
+}
```
Comment (ThisDrunkDane) [inline core/encoding/base64/base64.odin]: These probably shouldn't stick around in the actual `core` library

Comment (zhibog) [reply core/encoding/base64/base64.odin]: I can remove them. I had them for proof initially, since those are the test vectors from the actual RFC

Comment (ThisDrunkDane) [reply core/encoding/base64/base64.odin]: Hmm, you could also turn it into a usage example as you have in `csv` and then do that same for `base32`?

File: core/encoding/csv/csv.odin [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+write_raw :: proc(path: string, data: []byte) -> bool {
+    file, err := os.open(path, os.O_RDWR | os.O_CREATE | os.O_TRUNC);
+    if err != os.ERROR_NONE do return false;
+    defer os.close(file);
+
+    if _, err := os.write(file, data); err != os.ERROR_NONE do return false;
+    return true;
+}
+
+read :: proc(path: string, delimiter := DELIMITER, skip_header := false) -> ([]string, int, bool) {
```
Comment (ThisDrunkDane) [inline core/encoding/csv/csv.odin]: for some self documentation, perhaps give the return values names?

**Comment by ThisDrunkDane** [comment]
Have you thought about writing versions of the csv stuff that operates on strings as well? and not only on file paths. People might wanna load the csv data themselves into a string

File: core/encoding/csv/csv.odin (hunk 2 above)
Comment (zhibog) [reply core/encoding/csv/csv.odin]: Done

File: core/encoding/base32/base32.odin [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25,  0,  0,  0,  0,  0,
+	 0,  0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 
+    15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25,  0,  0,  0,  0,  0,
+	 0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  
+     0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+	 0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  
+     0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+	 0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  
+     0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0
+};
```
Comment (gingerBill) [inline core/encoding/base32/base32.odin]: Pretty pattern

File: core/encoding/base32/base32.odin [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+		switch len(data) {
+            case:
+                out[7] = ENC_TABLE[data[4] & 0x1f];
+                carry = data[4] >> 5;
+                fallthrough;
+            case 4:
+                out[6] = ENC_TABLE[carry | (data[3] << 3) & 0x1f];
+                out[5] = ENC_TABLE[(data[3] >> 2) & 0x1f];
+                carry = data[3] >> 7;
+                fallthrough;
```
Comment (gingerBill) [inline core/encoding/base32/base32.odin]: Please use tabs for indentation only.

File: core/encoding/base32/base32.odin (hunk 3 above)
Comment (zhibog) [reply core/encoding/base32/base32.odin]: Should be ok now.

File: core/encoding/base32/base32.odin (hunk 4 above)
Comment (zhibog) [reply core/encoding/base32/base32.odin]: I went through and I think I caught all of them

**Comment by ThisDrunkDane** [comment]
Can you open another one with just the Base32 stuff? :)

**Pull request merged**